server/aesdsocket.c

```diff
@@ -218,21 +232,29 @@ void *handle_client_connection(void *client_fd_ptr)
 
     while ((bytes_received = recv(client_fd, buffer, BUFFER_SIZE, 0)) > 0)
     {
+        syslog(LOG_USER, "Received data: %.*s\n", (int)bytes_received, buffer);
+
         pthread_mutex_lock(&file_mutex);
-        fwrite(buffer, 1, bytes_received, file);
-        fflush(file);
-        fseek(file, 0, SEEK_SET);
+        write(file_fd, buffer, bytes_received);
+        lseek(file_fd, 0, SEEK_SET);
```

Comment:
Try closing and reopening the file descriptor here instead of using lseek.  I think your file position is not being updated.